src/dbnode/persist/fs/cross_block_reader_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fs
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/x/ident"
+	xtest "github.com/m3db/m3/src/x/test"
+	xtime "github.com/m3db/m3/src/x/time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var expectedError = errors.New("expected error")
+
+func TestCrossBlockReaderRejectMisconfiguredInputs(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	dfsReader := NewMockDataFileSetReader(ctrl)
+	dfsReader.EXPECT().IsOrderedByIndex().Return(false)
+
+	_, err := NewCrossBlockReader([]DataFileSetReader{dfsReader})
+
+	assert.Equal(t, errReaderNotOrderedByIndex, err)
+}
+
+func TestCrossBlockReaderRejectMisorderedInputs(t *testing.T) {
+	ctrl := xtest.NewController(t)
+	defer ctrl.Finish()
+
+	now := time.Now().Truncate(time.Hour)
+	dfsReader1 := NewMockDataFileSetReader(ctrl)
+	dfsReader1.EXPECT().IsOrderedByIndex().Return(true)
+	dfsReader1.EXPECT().Range().Return(xtime.Range{Start: now})
+
+	later := now.Add(time.Hour)
+	dfsReader2 := NewMockDataFileSetReader(ctrl)
+	dfsReader2.EXPECT().IsOrderedByIndex().Return(true)
+	dfsReader2.EXPECT().Range().Return(xtime.Range{Start: later})
+
+	_, err := NewCrossBlockReader([]DataFileSetReader{dfsReader2, dfsReader1})
+
+	expectedErr := fmt.Errorf("dataFileSetReaders are not ordered by time (%s followed by %s)", later, now)
+
+	assert.Equal(t, expectedErr, err)
+}
+
+func TestCrossBlockReader(t *testing.T) {
+	tests := []struct {
+		name           string
+		blockSeriesIds [][]string
```

Comment:
nit: linter complains unless this is `blockSeriesIDs`

Reply:
👍